src/Http/Routing/src/HttpMethodMetadata.cs

```diff
@@ -41,10 +41,60 @@ public HttpMethodMetadata(IEnumerable<string> httpMethods, bool acceptCorsPrefli
                 throw new ArgumentNullException(nameof(httpMethods));
             }
 
-            HttpMethods = httpMethods.ToArray();
+            HttpMethods = httpMethods.Select(GetHttpMethod).ToArray();
             AcceptCorsPreflight = acceptCorsPreflight;
         }
 
+        private static string GetHttpMethod(string httpMethod)
+        {
+            if (Http.HttpMethods.IsGet(httpMethod))
```

Comment:
Why the `Http.` namespace prefix rather than a `using`?

Reply:
There's already a property named `HttpMethods` here.